beacon-chain/rpc/eth/beacon/sync_committee.go

```diff
@@ -20,64 +18,82 @@ import (
 // ListSyncCommittees retrieves the sync committees for the given epoch.
 // If the epoch is not passed in, then the sync committees for the epoch of the state will be obtained.
 func (bs *Server) ListSyncCommittees(ctx context.Context, req *eth.StateSyncCommitteesRequest) (*eth.StateSyncCommitteesResponse, error) {
-	var st state.BeaconState
-	if req.Epoch != nil {
-		slot, err := corehelpers.StartSlot(*req.Epoch)
-		if err != nil {
-			return nil, status.Errorf(codes.Internal, "Could not calculate start slot for epoch %d: %v", *req.Epoch, err)
-		}
-		st, err = bs.StateFetcher.State(ctx, []byte(strconv.FormatUint(uint64(slot), 10)))
-		if err != nil {
-			return nil, helpers.PrepareStateFetchGRPCError(err)
-		}
-	} else {
-		var err error
-		st, err = bs.StateFetcher.State(ctx, req.StateId)
-		if err != nil {
-			return nil, helpers.PrepareStateFetchGRPCError(err)
-		}
+	st, err := bs.stateFromRequest(ctx, &stateRequest{
+		epoch:   req.Epoch,
+		stateId: req.StateId,
+	})
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Could not fetch beacon state using request: %v", err)
+	}
+
+	// Get the current sync committee and sync committee indices from the state.
+	committeeIndices, committee, err := currentCommitteeIndicesFromState(st)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Could not get sync committee indices from state: %v", err)
+	}
+	subcommittees, err := extractSyncSubcommittees(st, committee)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Could not extract sync subcommittees: %v", err)
 	}
 
+	return &eth.StateSyncCommitteesResponse{
+		Data: &eth.SyncCommitteeValidators{
+			Validators:          committeeIndices,
+			ValidatorAggregates: subcommittees,
+		},
+	}, nil
+}
+
+func (bs *Server) SubmitSyncCommitteeSignature(ctx context.Context, message *eth.SyncCommitteeMessage) (*empty.Empty, error) {
+	panic("implement me")
+}
+
+func currentCommitteeIndicesFromState(st state.BeaconState) ([]types.ValidatorIndex, *ethpb.SyncCommittee, error) {
 	committee, err := st.CurrentSyncCommittee()
 	if err != nil {
-		return nil, status.Errorf(codes.Internal, "Could not get sync committee: %v", err)
+		return nil, nil, status.Errorf(
+			codes.Internal, "Could not get sync committee: %v", err,
```

Comment:
I think we should return regular errors from helper functions and `status` errors only from top-level functions.